test/e2e/clusterctl_upgrade.go

```diff
@@ -334,6 +336,18 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 
 		By("THE MANAGEMENT CLUSTER WITH THE OLDER VERSION OF PROVIDERS IS UP&RUNNING!")
 
+		machineCRD := &apiextensionsv1.CustomResourceDefinition{}
+		if err := managementClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "machines.cluster.x-k8s.io"}, machineCRD); err != nil {
+			Expect(err).ToNot(HaveOccurred(), "failed to retrieve a machine CRD")
+		}
+
+		// Build GroupVersionKind
+		machineGVK := schema.GroupVersionKind{
```

Comment:
Sorry, again nitpicking

```suggestion
		machineListGVK := schema.GroupVersionKind{
```